goon-universe/queries.js

```diff
@@ -539,33 +539,52 @@ function getStakeholderDescriptions(scenarioID){
     }
 }
 
+// Replace all these with a single helper taking an order parameter?
 function getInitReflectPage(scenarioID, callback){
-    let thisQuery= 'select prompt.prompt from pages, prompt where pages.id = prompt.page_id and pages.order = '+ INITIAL_REFLECTION +' and scenario_id = $1'
+    let thisQuery= 'select pages.body_text, prompt.prompt, prompt.prompt_num from pages, prompt where pages.id = prompt.page_id and pages.order = '+ INITIAL_REFLECTION +' and scenario_id = $1'
     pool.query(thisQuery, [scenarioID], (error,results) => {
         if (error) {
             throw error
         }
-        callback(results.rows)
+        let response = {}
+        response.prompts = results.rows.map(row => ({
+            text: row.prompt,
+            id: row.prompt_num
+        }))
+        response.body_text = results.rows[0].body_text
```

Comment:
The code works fine in most cases. However, when a scenario id that does not exist in the database is passed, the code crashes with a 'cannot read property body_text of undefined'. This needs to fixed. I think a simple if(results.length == 0) check will solve the problem

Reply:
Returning(callback) an empty object if that condition is true will be the best thing to do since it will work with the API without having to change anything